### PR TITLE
added checker for toBjj address in transaction creation

### PR DESCRIPTION
**Pull request by Mikelle**
Closes https://github.com/hermeznetwork/hermez-node/issues/739

### What does this PR does?

added checker for toBjj address in transaction creation

### CODE REVIEW

File: db/l2db/apiqueries.go [hunk 1]
```diff
@@ -69,6 +69,16 @@ func (l2db *L2DB) AddTxAPI(tx *PoolL2TxWrite) error {
 			feeUSD, l2db.maxFeeUSD))
 	}
 
+	if tx.ToBJJ != nil {
+		var exists bool
+		row := l2db.dbRead.QueryRow(`SELECT EXISTS(SELECT 1 FROM account WHERE bjj = $1);`, tx.ToBJJ)
+		if err := row.Scan(&exists); err != nil {
+			return tracerr.Wrap(err)
+		} else if !exists {
```
Comment (jeffprestes) [inline db/l2db/apiqueries.go:77]: Maybe split this if/else chain in two verifications would it make the code more clean. 
One if verifies the error
Other if, after the value is set to exists variable, check the exists variable content.

**Review by Pantani — CHANGES_REQUESTED** [review]
Let's wait for the answer to it:
https://github.com/hermeznetwork/hermez-node/issues/739#issuecomment-828370528

**Comment by jeffprestes** [comment]
After meeting with Jordi, Carlos and Bennassar we checked the user can send tokens to a BJJ that is not already created but internally the before the hermez-node create a new account it needs to check if the account already exists and reuse it. So, this implementations needs to be made in tx-selector or batch builder and not in API. The API must accept the tx.

**Comment by Pantani** [comment]
@jeffprestes @Mikelle, can we close this one?

**Comment by Mikelle** [comment]
@Pantani I think yes, this pr is no longer relevant, bcs API have to accept such transactions

**Pull request closed without merging**